api/src/main/java/org/apache/iceberg/expressions/ExpressionUtil.java

```diff
@@ -539,7 +539,8 @@ private static String sanitize(Type type, Object value, long now, int today) {
       case FIXED:
       case BINARY:
       case VARIANT:
-        // for boolean, uuid, decimal, fixed, variant, and binary, match the string result
+      case UNKNOWN:
+        // for boolean, uuid, decimal, fixed, variant, unknown, and binary, match the string result
         return sanitizeSimpleString(value.toString());
```

Comment:
Is this going to be an issue if unknown value as a non-valid string representation? I haven't looked to hard into the potential issues here but i'm imaging a situation where a binary ends up here and it doesn't have a UTF-8 representation? Maybe not possible here?

Reply:
I kind of feel like maybe Unknown should just output "Unknown" rather than anything sanitized?

Reply:
Yes, that's a good point. I've changed it to `(unknown)`, similar to `(time)`